### PR TITLE
TCPSocket could not be used as it was not checked in validation

### PR DESCRIPTION
**Pull request by smarterclayton**
Attempting to use it gave the error "must register one handler".
Added more tests for it.

### CODE REVIEW

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -454,6 +454,14 @@ func validateHTTPGetAction(http *api.HTTPGetAction) errs.ValidationErrorList {
 	return allErrors
 }
 
+func validateTCPSocketAction(tcp *api.TCPSocketAction) errs.ValidationErrorList {
+	allErrors := errs.ValidationErrorList{}
+	if tcp.Port.IntVal == 0 {
```
Comment (thockin) [inline pkg/api/validation/validation.go]: Do we want a comment about why this does not support named ports?  E.g. "Named ports are probably going to be deprecated, so don't handle that here."

Comment (smarterclayton) [inline pkg/api/validation/validation.go]: Ill just make it correct - typo on my part

**Comment by smarterclayton** [comment]
Comment addressed

**Comment by thockin** [comment]
LGTM

**Comment by zmerlynn** [comment]
This raced with #4927 and broke the build. Reverting this one arbitrarily.

**Pull request merged**